plugins/toolkit/jetbrains-core/src/software/aws/toolkits/jetbrains/services/codewhisperer/codescan/CodeWhispererCodeScanManager.kt

```diff
@@ -454,21 +459,31 @@ class CodeWhispererCodeScanManager(val project: Project) {
             val editorFactory = EditorFactory.getInstance()
             editorFactory.eventMulticaster.addDocumentListener(documentListener, project)
             editorFactory.addEditorFactoryListener(fileListener, project)
+
+            // Add document listeners to all files under project root for files of supported languages.
+            addListeners()
         }
     }
 
     private fun addListeners() {
-        fileNodeLookup.keys.forEach { file ->
+        val projectRoot = project.basePath?.let { Path.of(it) }?.toFile()
+        val files = projectRoot?.listFiles()
+        val localFileSystem = LocalFileSystem.getInstance()
+        files?.forEach { file ->
             runInEdt {
-                val document = FileDocumentManager.getInstance().getDocument(file)
-                if (document == null) {
-                    LOG.error { message("codewhisperer.codescan.file_not_found", file.path) }
-                    return@runInEdt
+                try {
+                    localFileSystem.findFileByIoFile(file)?.let { virtualFile ->
+                        if (!virtualFile.isDirectory && virtualFile.programmingLanguage().toTelemetryType() != CodewhispererLanguage.Unknown &&
+                            virtualFile.programmingLanguage().toTelemetryType() != CodewhispererLanguage.Plaintext
+                        ) {
+                            virtualFile.getDocument().addDocumentListener(documentListener, project)
+                        }
+                    }
+                } catch (e: Exception) {
+                    LOG.debug(e) { "Failed to add document listener to file: $file" }
```

Comment:
have we tested this in a large project?

Reply:
changed this add Listeners to add them only for open files and files with issues.